test/extended/util/cloud/cloud.go

```diff
@@ -43,65 +45,76 @@ func (c *ClusterConfiguration) ToJSONString() string {
 	return string(out)
 }
 
-// LoadConfig uses the cluster to setup the cloud provider config.
-func LoadConfig(clientConfig *rest.Config) (*ClusterConfiguration, error) {
-	// LoadClientset but don't set the UserAgent to include the current test name because
-	// we don't run any test yet and this call panics
-	coreClient, err := clientset.NewForConfig(clientConfig)
-	if err != nil {
-		return nil, err
-	}
-	configClient, err := configclient.NewForConfig(clientConfig)
-	if err != nil {
-		return nil, err
-	}
-	operatorClient, err := operatorclient.NewForConfig(clientConfig)
-	if err != nil {
-		return nil, err
-	}
+// DiscoverConfig uses the cluster to setup the cloud provider config.
+func DiscoverConfig() (*ClusterConfiguration, error) {
+	p, masters, nonMasters, networkSpec := testPlatformStatus, testMasters, testNonMasters, testNetworkSpec
```

Comment:
No, please no. No globals.  If you want to preserve stuff, keep it in a struct you return from this package.